source/mir/random/discrete.d

```diff
@@ -135,9 +137,10 @@ struct Discrete(T)
             arr[sm.index] = AltPair(sm.prob, lg.index);
 
             // update larger & reinsert
-            lg.prob += sm.prob - 1;
+            lg.prob += sm.prob;
+            lg.prob -= 1;
 
-            if (lg.prob < 1)
+            if (lg.prob < 1 || lg.prob.approxEqual(1, 1e-13, 1e-13))
```

Comment:
This was added to increase the stability of the Discrete sampling when seeded with a static seed, but using different FP maths (e.g. `double` vs. `real`). I think it's a fair assumption to say that everything that's closely 0 is zero?